src/MarketPlace.sol

```diff
@@ -165,39 +152,22 @@ contract MarketPlace {
         );
 
         {
-            // create the principal tokens array
-            address[9] memory market = [
-                illuminateToken, // Illuminate
-                t[0], // Swivel
-                t[1], // Yield
-                t[2], // Element
-                t[3], // Pendle
-                t[4], // Tempus
-                t[5], // Sense
-                t[6], // APWine
-                t[7] // Notional
-            ];
-
-            // Set the market
-            markets[u][m] = market;
-
-            // Have the lender contract approve the several contracts
-            ILender(lender).approve(u, a, e, t[7], sensePeriphery);
-
-            // Allow converter to spend interest bearing asset
-            if (t[5] != address(0)) {
-                IRedeemer(redeemer).approve(h);
-            }
-
-            // Approve interest bearing token conversion to underlying for APWine
-            if (t[6] != address(0)) {
-                address futureVault = IAPWineToken(t[6]).futureVault();
-                address interestBearingToken = IAPWineFutureVault(futureVault)
-                    .getIBTAddress();
-                IRedeemer(redeemer).approve(interestBearingToken);
+            // assign values for the principal tokens and adapters array
+            for (uint i = 0; i < t.length; i++) {
+                markets[u][m][i + 1] = t[i];
+                adapters[u][m][i + 1] = a[i];
             }
+        }
 
-            emit CreateMarket(u, m, market, e, a);
+        {
+            address underlying = u;
+            uint256 maturity = m;
```

Comment:
Does it actually reduce the stack somehow by storing these rather than just inputting them in the event? 

If so this sort of optimization always confuses me

Reply:
Yes it does. I'm not sure what the mechanism is behind why this works, but it is necessary to compile the contract. I'll ask around to find out more, but I suspect it has something to do with how the optimizer parses the AST when it enters new scopes. My leading theory is formally reassigning the calldata helps the optimizer throw out other parameters from the calldata that it otherwise was undecided on, and in doing so, reduces the stack during compile time.